sycl/include/sycl/accessor.hpp

```diff
@@ -236,7 +236,8 @@ template <typename... Ts>
 void cannot_be_called_on_host([[maybe_unused]] const char *API,
                               Ts &&.../* ignore */) {
 #ifndef __SYCL_DEVICE_ONLY__
-  std::fprintf(stderr, "%s cannot be called on host!\n", API);
+  fputs(API, stderr);
+  fputs(" cannot be called on host!\n", stderr);
```

Comment:
Why can't we use `std::cerr` here?

Reply:
Don't want to deal with C++ streams

Reply:
Sorry, but why? IIUC, it's host code. We have used `std::cout`/`std::cerr` at other places well in RT

Reply:
Hmm, I vaguely remember that it gave me some issues but can't remember what. Switching to `std::cerr` (PR description updated as well). Let's see how CI likes it.